### PR TITLE
Adding support from focal to noble

### PR DESCRIPTION
**Pull request by addyess**
## Overview 
To support running on `noble` and `jammy` while still building/running on `focal` update python charm libaries

## Details
* Update metadata to support noble by moving metadata to charmcraft.yaml
* update charm systemd library for noble support

### CODE REVIEW

**Comment by addyess** [comment]
Draft waiting on https://github.com/canonical/charms.reactive/pull/260 to be pushed to pypi

**Comment by addyess** [comment]
Waiting on an [answer] regarding changing the `assumes` in charmcraft.yaml

[answer]: https://matrix.to/#/!xzmWHtGpPfVCXKivIh:ubuntu.com/$qJ4QG0vkPqjXNM85e1bO3If04HN2DjutqJblrZewGjQ?via=ubuntu.com&via=matrix.org&via=matrix.debian.social

File: src/nginx.py [hunk 1]
```diff
@@ -53,17 +55,21 @@ def defaults(self) -> dict:
...
         """
         return {k: self.custom.get(k, v) for k, v in self.charm_default.items()}
 
     @property
     def extra(self) -> dict:
         """Returns the configurations that are exclusively in the custom configuration.
 
-        Returns:
+        Returns
+        -------
```
Comment (mateoflorido) [inline src/nginx.py:70]: Is this what the cool kids use these days?

**Review by mateoflorido — APPROVED** [review]
LGTM!

File: src/nginx.py (hunk 1 above)
Comment (addyess) [reply src/nginx.py:70]: must be from `ruff`?  I don't remember adding these myself

**Pull request merged**